packages/components/nodes/chains/GraphCypherQAChain/GraphCypherQAChain.ts

```diff
@@ -0,0 +1,273 @@
+                optional: true,
+                description: 'Model for generating Cypher queries and answers.'
+            },
+            {
+                label: 'Cypher Generation Model',
+                name: 'cypherModel',
+                optional: true,
+                type: 'BaseLanguageModel',
+                description: 'Model for generating Cypher queries. If not provided, the main model will be used.'
+            },
+            {
+                label: 'QA Model',
+                name: 'qaModel',
+                optional: true,
+                type: 'BaseLanguageModel',
+                description: 'Model for generating answers. If not provided, the main model will be used.'
+            },
+            {
+                label: 'Neo4j Graph',
+                name: 'graph',
+                type: 'Neo4j'
+            },
+            {
+                label: 'Memory',
+                name: 'memory',
+                type: 'BaseMemory',
+                optional: true,
+                description: 'Memory to store chat history for context'
+            },
+            {
+                label: 'Cypher Generation Prompt',
+                name: 'cypherPrompt',
+                type: 'BasePromptTemplate',
+                description: 'Prompt template for generating Cypher queries. Must include {schema} and {question} variables'
+            },
+            {
+                label: 'QA Prompt',
+                name: 'qaPrompt',
+                optional: true,
+                type: 'BasePromptTemplate',
+                description: 'Prompt template for generating answers. Must include {context} and {question} variables'
+            },
+            {
+                label: 'Input Moderation',
+                description: 'Detect text that could generate harmful output and prevent it from being sent to the language model',
+                name: 'inputModeration',
+                type: 'Moderation',
+                optional: true,
+                list: true
+            },
+            {
+                label: 'Return Direct',
+                name: 'returnDirect',
+                type: 'boolean',
+                default: false,
+                optional: true,
+                description: 'If true, return the raw query results instead of using the QA chain'
+            },
+            {
+                label: 'Return Intermediate Steps',
+                name: 'returnIntermediateSteps',
+                type: 'boolean',
+                default: false,
+                optional: true,
+                description: 'If true, return the intermediate steps in the chain execution'
+            }
+        ]
+        this.outputs = [
+            {
+                label: 'Graph Cypher QA Chain',
+                name: 'graphCypherQAChain',
+                baseClasses: [this.type, ...getBaseClasses(GraphCypherQAChain)]
+            },
+            {
+                label: 'Output Prediction',
```

Comment:
here we allow this chain to output a string or json, but we dont have any function that checks for the output

You can reference [LLMChain](https://github.com/FlowiseAI/Flowise/blob/main/packages/components/nodes/chains/LLMChain/LLMChain.ts#L112)

Reply:
I took RetrievalQAChain as a reference and removed Output Prediction

Reply:
I have added it back. Works now!